File: 0011-container-with-most-water/0011-container-with-most-water.c
int maxArea(int* height, int heightSize) {
    int left = 0, right = heightSize - 1;
    int area = 0;

    while(left < right) {
        int currArea = 0;
        if(height[left] < height[right]) {
            currArea = height[left] * (right - left);
            left++;
        } else {
            currArea = height[right] * (right - left);
            right--;
        }
        if(currArea > area) area = currArea;
    }

    return area;
}

File: 0011-container-with-most-water/0011-container-with-most-water.cs
public class Solution {
    public int MaxArea(int[] height) {
        int left = 0, right = height.Length - 1, area = 0;

        while(left < right) {
            if(height[left] < height[right]) {
                area = area < height[left] * (right - left) ? height[left] * (right - left) : area;
                left++;
            } else {
                area = area < height[right] * (right - left) ? height[right] * (right - left) : area;
                right--;
            }
        }

        return area;
    }
}